### ISSUE TITLE
Add properties required by FDA certification

### ISSUE THREAD

**Issue by AlexanderYukhanov** [CycloneDX/cyclonedx-property-taxonomy#104]
FDA requires (https://www.fda.gov/media/119933/download) suppliers to provide two additional attributes:
"
- The software level of support provided through monitoring and maintenance from the
software component manufacturer (e.g., the software is actively maintained, no longer
maintained, abandoned); and
- The software component’s end-of-support date. 
"

It would be beneficial to create taxonomy containing 4 additional attributes to meet this requirement:
fda:lifecycle:suport_level
fda:lifecycle:suport_level_comment
fda:lifecycle:end_of_support
fda:lifecycle:end_of_support_comment

The comment attributes are required to provide the justification for the provided attributes or the conducted effort description if those attributes were not detected

**Comment by AlexanderYukhanov** [comment]
No, it's required to provide the level of support and justification both for level of support and the end of support date. Probably, we can extend cdx:lifecycle instead of declaring new fda:lifecycle like

cdx:lifecycle:suportLevel
cdx:lifecycle:suportLevelComment or cdx:lifecycle:suportLevelJustification
cdx:lifecycle: endOfSupportComment or cdx:lifecycle: endOfSupportJustification
?

**Comment by stevespringett** [comment]
Is there a definition of support level?

See also: [Common Lifecycle Enumeration](https://docs.google.com/document/d/1sRMS1IX0r7ZkYthDR0VY1bYyvp_6K_xw4sR1vZwla8E/edit).

**Comment by AlexanderYukhanov** [comment]
No, there is no formal definition or additional guidance from the FDA on this. That's why including a justification field alongside the support level is beneficial. It allows vendors to apply their own criteria (for example, using the OSSF Scorecard's 'Maintained' score).

The Common Lifecycle Enumeration currently covers different use-cases and is not well-suited for describing the support levels of dependencies, especially OSS one.

**Comment by jkowalleck** [comment]
a solution seams to be discussed here: https://github.com/CycloneDX/specification/issues/400

**Comment by shoeffner** [comment]
For reference, the FDA certification requires the following (cf. page 18 https://www.fda.gov/media/119933/download):

> In addition to the minimum elements identified by NTIA, for each software component
> contained within the SBOM, manufacturers should include in the premarket submission:
> - The software level of support provided through monitoring and maintenance from the
>    software component manufacturer (e.g., the software is actively maintained, no longer
>    maintained, abandoned); and
> - The software component’s end-of-support date.
> 
> When provided, manufacturers may choose to provide these additional elements as part of the
> SBOM, or they may provide it separately, such as in an addendum. Industry-accepted formats of
> SBOMs are encouraged.
> If a manufacturer is unable to provide the SBOM information to FDA, the manufacturer should
> provide a justification for why the information cannot be included in the premarket submission.

The _end-of-support date_ can be specified with [`cdx:lifecycle:milestone:endOfSupport`](https://github.com/CycloneDX/cyclonedx-property-taxonomy/blob/479acf89cc0595995ad20d7fa9439f007250b050/cdx/lifecycle.md#cdxlifecyclemilestone-namespace-taxonomy), as it is indeed a specific date.

It is more difficult with the "level of support", as the FDA did not (at least not in the linked document) specify which values are acceptable; from the text it seems that at least:
- actively maintained
- no longer maintained
- abandoned

should be fine, although I am not even sure if I can tell a semantic difference between "no longer maintained" and "abandoned".

The document [Common Lifecycle Enumeration](https://docs.google.com/document/d/1sRMS1IX0r7ZkYthDR0VY1bYyvp_6K_xw4sR1vZwla8E/edit) only specifies specific dates (which are the base for the `cdx:lifecycle` specs), but not the "textual" or "semantic" form of "level of support".

In https://github.com/CycloneDX/specification/issues/400, @sjn has a few interesting ideas in https://github.com/CycloneDX/specification/issues/400#issuecomment-2202190961; I especially think the list [Project Support Indicators](https://github.com/CPAN-Security/security.metacpan.org/blob/supplychain-sbom/docs/foss-project-lifecycle.md#project-support-indicators) might be a very good start for the level of support as required by the FDA.

Something like that could be added to the cdx:lifecycle namespace, e.g.:

> # `cdx:lifecycle` Namespace Taxonomy
> 
> | Namespace | Description |
> |-----------|-------------|
> | `cdx:lifecycle:support` | Namespace for milestone-specific information. |
> 
> ## `cdx:lifecycle:support` Namespace Taxonomy
> 
> | Property | Description |
> |----------|-------------|
> | `cdx:lifecycle:support:level` |  One of `MAINTAINED`, `CASUAL`, `DONE`, `DEPRECATED`, `SECURITYONLY`, `SUPERSEDED`, `UNMAINTAINED`. etc. |
> | `cdx:lifecycle:support:levelReason` | A custom string explaining the reason why the support was rated as-is |

Of course, this is just an idea and it does not really cover the requirement to justify why there is no known end-of-support date, for example (which is often the case for small to medium Open Source projects, as often do not have any guarantees).

Otherwise, I totally agree that something like an `fda:` (or, for https://github.com/CycloneDX/specification/issues/400, `ce:`) namespace could also be an option.

**Comment by AlexanderYukhanov** [comment]
I think when it comes to the compliance, it makes sense to put the FDA requested information into a single place. So cdx:lifecycle:support with level, levelReason, endOfSupportDate and endOfSupportDateReason sound like a better solution to me. Here endOfSupportDateReason is needed to describe the details or the description why the endOfSupportDate is not provided.

**Comment by shoeffner** [comment]
I am not sure it has to, especially since the forms in eSTAR (the FDAs submission platform, as I have been told by a colleague working on that) seem to expect a second document:

> - Please attach your Software Bill of Materials (SBOM)
> - Please attach a document to provide the software level of support and end-of-support date for each software component (e.g. OTS software) identified in the SBOM. For any component where this information was not available, provide a justification.

So in general, it seems not necessarily a requirement to have that information inside the SBOM, although convenient.

**Comment by braggpeaks** [comment]
Despite not specifically requested by FDA, the information is totally suited for the SBOM and I'd find it good to have this as cdx:lifecycle:support - @jkowalleck can @shoeffner 's proposal be added to the [lifecycle namespace taxonomy](https://github.com/CycloneDX/cyclonedx-property-taxonomy/blob/main/cdx/lifecycle.md)?

**Comment by jkowalleck** [comment]
>  Maybe someone with connections could ask around at the FDA how this should be handled.

TBH, I would really love to see US FDA to register and maintain their property taxonomy according to their needs themselves.
Just like the German Federal Office for Information Security did - they registered the TL-namespace `bsi` and maintain it themselves.

**Comment by braggpeaks** [comment]
One of the questions we got back from our FDA submission was exactly about this topic. Will keep you posted about it.

Regarding the level of support, it is not uncommon that outdated software components will kept on being used, e.g. old Windows OS on ATMs. I’ve frequently seen this in the manufacturing of medical products as well. Since obtaining product clearance is both costly and time-consuming, manufacturers tend to continue production without modifications once a product is approved. From a cybersecurity standpoint, this is problematic, but it is a common industry practice. Requiring manufacturers to include a Software Bill of Materials (SBOM) in their submissions would make it immediately clear where outdated software components are being used. This transparency could later serve as a basis for rejecting approvals.

**Comment by sjn** [comment]
> I believe this will require further discussion, as my proposal is not fully fledged out – I disagree with some of the "level of support" classifications by CPAN, for example, but it was a good first idea.

Just for clarification; The "support levels" I've been talking about in https://github.com/CPAN-Security/security.metacpan.org/blob/supplychain-sbom/docs/foss-project-lifecycle.md#project-support-indicators and https://github.com/CycloneDX/specification/issues/400 aren't on CPAN, though they may be _inspired_ by it.

For me, the important point with these indicators is: "What does the project _maintainer_ promise?", and leave it up to the reader to make a judgment on whether this is good enough for them.

For example, if a maintainer says they only offer "CASUAL" support (meaning, they might not be responsive to emergencies, and only reply to issues or requests every other weekend when they don't have to tend to their kids), then this might be a signal that the user/manufacturer/business should explore some of the options they have to improve this situation. (e.g. fund the project; do their own due diligence; fork the project; offer to become co-maintainer; or after a license review, vendor the relevant parts of the code into their own product).

My hope with the "level of support" classifications I've put together, is to have enough of then to cover the most common scenarios for the _maintainer_ to choose between. It is, after all, their project and their time the rest of us are beholden to.

**Comment by sjn** [comment]
By the way, we'll probably be discussing these issues in the coming meetings in the OSS Sustainability WG; [Feel free to join](https://cyclonedx.org/participate/working-groups/)! 😁

**Comment by jorlando-elucid** [comment]
I handled this in https://github.com/CycloneDX/cyclonedx-property-taxonomy/issues/185.

https://github.com/CycloneDX/cyclonedx-property-taxonomy/issues/104#issuecomment-2688163620 This may be the right answer logically, but we all know the current and presumed future USA administrations aren't about to wake up one day and be generally competent. Having official support for these fields would be huge for us in regulated industries.

**Comment by jkowalleck** [comment]
not quite sure about CycloneDX CoreWorkingGroup maintaining this taxonomy for FDA.
The orgs could maintain this list themselves. Just like BSI maintains for their BSI [TR-03183](https://www.bsi.bund.de/EN/Themen/Unternehmen-und-Organisationen/Standards-und-Zertifizierung/Technische-Richtlinien/TR-nach-Thema-sortiert/tr03183/TR-03183_node.html) themselves in https://github.com/BSI-Bund/tr-03183-cyclonedx-property-taxonomy
